core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -62,7 +62,12 @@ public static Object getMeasureValueBasedOnDataType(String msrValue, DataType da
       case INT:
         return Double.valueOf(msrValue).longValue();
       case LONG:
-        return Long.valueOf(msrValue);
+        int index = msrValue.indexOf(".");
+        if (index != -1) {
+          return Long.valueOf(msrValue.substring(0,index));
+        } else {
+          return Long.valueOf(msrValue);
+        }
```

Comment:
@manishgupta88 Hi, in this PR, I want to solve the problem described as below:

when the Loding msrValue is decimal like "3.1415926" and the column type is Long, the original flow will not work well. It will return us a null value.